### DISCUSSION TITLE
Feature Request: Visual Studio extension for Mac

### DISCUSSION THREAD

**Discussion by rchoudhary** [AvaloniaUI/Avalonia#9338] [Q&A] [answered]
I had to use XAML/UWP for an internship and it's my favorite way of making a GUI. I found this framework and was excited to try it out, however I have a Mac at home. The extension only comes in `.vsix` format which isn't supported by Visual Studio for Mac. If a corresponding plugin for Mac were made that would be great!

**Comment by wdcossey** [comment]
According to the documentation the Avalonia (solution) doesn't work in Visual Studio for Mac (formerly Xamarin studio), I tried this myself a few days ago.

I did however build and run a project (not the source) using the NuGet packages in Rider (by JetBrains), it's worth noting there's no Visualizer (xaml preview).

I'm sure the devs are looking into VS for Mac, they might give you an update or eta.

**Comment by danwalmsley** [comment]
We also have AvalonStudio ide which is soon going to be released it gives you a previewer on mac, but you will have to wait for new mac backend for avalonia to be released first.

**Comment by danwalmsley** [comment]
https://github.com/AvaloniaUI/Avalonia/pull/1789

**Comment by danwalmsley** [comment]
https://github.com/vitalelement/avalonstudio

**Comment by wdcossey** [comment]
I have had a quick glance at Avalon studio, it's not a bad effort.

I wouldn't want to discourage the developers from making their own IDE but I think supporting existing IDE's (VS and Rider) would be more beneficial than making your own as these are well established and feature rich products.

That said, I'll have a gander at the source code and see if I can make any contributions or suggestions.

**Comment by danwalmsley** [comment]
This is a side project that has different goals, plugins for other ides are still wanted.

**Comment by kekekeks** [comment]
The problem with VSforMac is that while being based on MonoDevelop, it doesn't upstream the required changes back. So things like .NET Core debugger aren't available in MonoDevelop at all.

So it makes a VSforMac extension to be an effort that only covers one platform. We'd rather invest in VSCode or Rider extensions, since those IDEs work everywhere.

We are currently collaborating with JB folks to make Rider extension possible.

**Comment by Tylertron1998** [comment]
Hi there! I've been really interested in avalonia for a while. I'm a Rider user - and wondering, is there any update on the plugin? Do you have a beta version to use?

**Comment by Tylertron1998** [comment]
@kekekeks sorry for the mention. Can I get an update on the above please? (:

**Comment by worldbeater** [comment, marked as answer]
@Tylertron1998 please check the [AvaloniaRider](https://github.com/fornever/avaloniarider) repository. We are looking forward for more contributors to help out with that plugin btw, so please feel free to join and contribute! :1st_place_medal:

**Comment by timunie** [comment]
Converted this into Q&A as this is neither a bug nor a feature request. If you disagree please let us know.
